gateway/gateway.go

```diff
@@ -17,34 +16,110 @@ type Config struct {
 	Headers map[string][]string
 }
 
-// API defines the minimal set of API services required for a gateway handler.
+// ImmutablePath TODO: This isn't great, as its just a signal and mutable paths fulfill the same interface
+type ImmutablePath interface {
```

Comment:
Why `interface` ?

Reply:
Was matching the other path types, doesn't have to be though. This was mostly a placeholder